### PR TITLE
fix WeightedScheduler reporting incorrect queue cylces

### PR DESCRIPTION
**Pull request by R9295**
WeightedScheduler would never report a complete cycle since the runs would always be set to zero before it got the chance to surpass the corpus count. This PR fixes it

### CODE REVIEW

File: libafl/src/schedulers/weighted.rs [hunk 1]
```diff
@@ -324,13 +324,13 @@ where
 
             let wsmeta = state.metadata_mut::<WeightedScheduleMetadata>()?;
 
-            let current_cycles = wsmeta.runs_in_current_cycle();
+            let runs_in_current_cycle = wsmeta.runs_in_current_cycle();
 
             // TODO deal with corpus_counts decreasing due to removals
-            if current_cycles >= corpus_counts {
+            if runs_in_current_cycle >= corpus_counts {
                 wsmeta.set_runs_current_cycle(0);
```
Comment (R9295) [inline libafl/src/schedulers/weighted.rs:331]: The issue

File: libafl/src/schedulers/weighted.rs [hunk 2]
```diff
@@ -340,7 +340,7 @@ where
             };
 
             // Update depth
-            if current_cycles > corpus_counts {
+            if runs_in_current_cycle == corpus_counts {
```
Comment (tokatoka) [inline libafl/src/schedulers/weighted.rs:343]: i think >= is better? addison added that check because corpus_counts could actually decrease..

Comment (tokatoka) [reply libafl/src/schedulers/weighted.rs:343]: what i mean is that you can get corpus_counts decreased by N so runs_in_current_cycle is not necessarily equal to corpus count but you still have to update it.

Comment (R9295) [reply libafl/src/schedulers/weighted.rs:343]: Ah, makes sense.

**Comment by domenukk** [comment]
First letter of the PR description should be uppercase in this repo ;)

**Comment by domenukk** [comment]
Thanks!

**Pull request merged**